### PR TITLE
Modrs 118

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MODRS-118](https://issues.folio.org/browse/MODRS-118) - mod-remote-storage: folio-spring-base update

## Purpose
Upgrade folio-spring-base to 4.1.0
Upgrade spring-boot-starter-parent to 2.7.*
Remove unnecessary properties in pom.xml

## Approach
Change pom.xml.

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [x] Code coverage on new code is 80% or greater
   - [x] Duplications on new code is 3% or less
   - [x] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [x] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by siarhei-charniak — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -27,16 +27,13 @@
     <openapi-generator.version>5.3.1</openapi-generator.version>
     <mapstruct.version>1.4.1.Final</mapstruct.version>
     <json.version>20200518</json.version>
-    <hazelcast.version>4.0.2</hazelcast.version>
     <spring-kafka.version>2.8.2</spring-kafka.version>
```
Comment (julianladisch) [inline pom.xml:30]: This lines downgrades spring-kafka from 2.8.6 to 2.8.2.

Try `mvn dependency:tree -Dincludes=org.springframework.kafka:spring-kafka`:

```
[INFO] org.folio:mod-remote-storage:jar:1.6.0-SNAPSHOT
[INFO] \- org.springframework.kafka:spring-kafka:jar:2.8.2:compile
```

After removing `<spring-kafka.version>2.8.2</spring-kafka.version>`:

```
[INFO] org.folio:mod-remote-storage:jar:1.6.0-SNAPSHOT
[INFO] \- org.springframework.kafka:spring-kafka:jar:2.8.6:compile
```

File: pom.xml [hunk 2]
```diff
@@ -27,16 +27,13 @@
...
     <wiremock.version>2.27.2</wiremock.version>
     <spring-kafka-test.version>2.8.2</spring-kafka-test.version>
```
Comment (julianladisch) [inline pom.xml:32]: This line downgrades spring-kafka-test from 2.8.6 to 2.8.2 if a spring-kafke-test dependency gets added in future. Please remove this line.

File: pom.xml [hunk 3]
```diff
@@ -27,16 +27,13 @@
...
     <embedded.db.version>2.1.1</embedded.db.version>
     <pubsub.client.version>2.0.0</pubsub.client.version>
     <jackson.version>2.11.1</jackson.version>
     <json.path.version>2.5.0</json.path.version>
```
Comment (julianladisch) [inline pom.xml:36]: This line downgrades json-path from 2.7.0 to 2.5.0.

Try `mvn dependency:tree -Dincludes=com.jayway.jsonpath:json-path`:

```
[INFO] org.folio:mod-remote-storage:jar:1.6.0-SNAPSHOT
[INFO] \- com.jayway.jsonpath:json-path:jar:2.5.0:compile
```

After removing

```
<json.path.version>2.5.0</json.path.version>
```
and
```
<version>${json.path.version}</version>
```

running `mvn dependency:tree -Dincludes=com.jayway.jsonpath:json-path` yields

```
[INFO] org.folio:mod-remote-storage:jar:1.6.0-SNAPSHOT
[INFO] \- com.jayway.jsonpath:json-path:jar:2.7.0:compile
```

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Pull request merged**